keras_nlp/samplers/sampler.py

```diff
@@ -189,14 +189,27 @@ def _pad_prompt(self, prompt, max_length):
 
     def _mask_tokens_after_end_token(
         self,
-        prompt,
+        generated_result,
+        original_prompt,
         max_length,
         end_token_id,
     ):
         """Helper function to truncate the tokens after the end token."""
+        # Exclude original prompts from being truncated.
+        # Add a big int to tokens in original prompt to move token ids over
+        # the vocab size. Vocab size should not be large as 1e9.
+        increment_value = int(1e9)
```

Comment:
these feels a little hacky. let's try to think of a more elegant way to do this. maybe pass in the original mask after padding and do something like this...

```python
# Create a tensor with True for each end_token_id
end_tokens = generated_result == end_token_id
# Remove all end_token_ids in the original input.
end_tokens = end_tokens & (original_padded_mask == False)
# Find index of first end_token_id.
end_indices = tf.math.argmax(generated_result == end_token_id, -1)
```